lib/internal/test_runner/test.js

```diff
@@ -204,6 +210,12 @@ class Test extends AsyncResource {
       this.testNumber = parent.subtests.length + 1;
       this.timeout = parent.timeout;
       this.root = parent.root;
+      this.hooks = {
+        before: [],
+        after: [],
+        beforeEach: parent.hooks.beforeEach.slice(),
+        afterEach: parent.hooks.afterEach.slice(),
```

Comment:
```suggestion
        beforeEach: ArrayPrototypeSlice(parent.hooks.beforeEach),
        afterEach: ArrayPrototypeSlice(parent.hooks.afterEach),
```
assuming this can run after user code

Reply:
Does this work for something like this:

```js
describe('Root', () => {
  describe('Nested', () => {
    it('should run beforeEach in first test', () => {
      // always pass
    })

    it('should run beforeEach in second test', () => {
      // always pass
    })

    it('should run beforeEach in third test', () => {
      // always pass
    })
  })

  beforeEach(() => {
    console.log('test') // only prints when the Nested context is entered
  })
})

```

By the time that `beforeEach` is called the hooks would already have been copied.

Reply:
How do global hooks work too? There is no Test or Suite to attach those to 🤔

Reply:
sure, but the user could have invoked code before any `describe`s, or inside one but before an `it`.

Reply:
@ljharb To be clear, my above comments were about the original code, not about your suggestion.

Reply:
@MoLow Since the above was not addressed, should I open 2 other issues instead?